contentcuration/contentcuration/viewsets/channelset.py

```diff
@@ -50,11 +52,22 @@ class Meta:
         list_serializer_class = BulkListSerializer
 
 
+class ChannelSetFilter(FilterSet):
+
+    class Meta:
+        model = ChannelSet
+        fields = (
+            "editors",
+        )
+
+
 class ChannelSetViewSet(ValuesViewset):
     queryset = ChannelSet.objects.all()
     serializer_class = ChannelSetSerializer
+    filter_backends = (DjangoFilterBackend,)
+    filter_class = ChannelSetFilter
     permission_classes = [IsAuthenticated]
-    values = ("id", "name", "description", "channels", "secret_token__token")
+    values = ("id", "name", "description", "channels", "secret_token__token", "editors")
```

Comment:
There is an interesting gotcha here, where adding `editors` to the `values` call, because `editors` is a many to many field, if there is more than one editor, then this will cause the same channelset to be returned multiple times in the resultant query (each with a different `editors` value corresponding to each editor).

Although currently we have no way in the interface of adding additional editors to a collection, I imagine we will do so at some point in the future.

I would instead do this consistently with how we filter Channels, and instead annotate an `edit` flag on the viewset, which is then different on a per user basis.

This is annotated onto the Channels queryset here: https://github.com/learningequality/studio/blob/develop/contentcuration/contentcuration/models.py#L780 (this method is called for getting the queryset inside these viewsets, so doing something parallel in the method on ChannelSet would mean this annotation can be used in the FilterSet and returned as a value).

Reply:
Thanks for the feedback - I've made some updates based on these notes and our conversation yesterday, and I think it's now using the approach you suggest. Let me know if you have any other comments!